### PR TITLE
models/krate: Improve feature validation

### PR DESCRIPTION
**Pull request by Turbo87**
As promised in https://github.com/rust-lang/crates.io/pull/7500#pullrequestreview-1732333754, I have a few suggestions on how the feature validation code can be improved :)

### CODE REVIEW

File: src/models/krate.rs [hunk 1]
```diff
@@ -528,6 +511,34 @@ impl CrateVersions for [Crate] {
...
+    )]
+    Start(char, String),
+    #[error(
+        "invalid character `{0}` in feature `{1}`, characters must be Unicode \
+        XID characters, `+`, `-`, or `.` (numbers, `+`, `-`, `_`, `.`, or most \
+        letters)"
+    )]
+    Char(char, String),
+    #[error(transparent)]
+    DependencyName(#[from] InvalidDependencyName),
```
Comment (hi-rustin) [inline src/models/krate.rs:531]: I'm uncertain about placing it here, as it is also used in validate_dependency. It's not intended for validating the feature name. But I can help to improve it in my next PR.

**Review by hi-rustin — APPROVED** [review]
Looks good to me! Thanks!

File: src/models/krate.rs (hunk 1 above)
Comment (Turbo87) [reply src/models/krate.rs:531]: InvalidDependencyName can also be used directly :)

Comment (hi-rustin) [reply src/models/krate.rs:531]: Oh. Got it. It's nice. Thanks!

**Pull request merged**